io-engine/src/bdev/mod.rs

```diff
@@ -48,6 +50,30 @@ pub trait GetName {
     fn get_name(&self) -> String;
 }
 
+/// Check if the path indicates a blockdev, or a regular file. It will be
+/// helpful for cases where we use a regular file e.g. a tmpfs file, as a pool
+/// backend. XXX: If we can't clearly determine if a path is blockdev(error
+/// conditions), we return false today so that caller side can decide what to
+/// do.
+pub fn path_is_blockdev(path: &str) -> bool {
+    // Create a mutable stat struct
+    let mut stat_result: stat = unsafe { std::mem::zeroed() };
```

Comment:
> Why are you using unsafe here? Is it not possible to achieve the same using safe std api?

I don't see any, since `stat` is an external libc type here. This function's usage is widespread in Mayastor code.

Reply:
std::fs::metadata(path)?.is_block_device() ?

Reply:
> std::fs::metadata(path)?.is_block_device() ?

Looks relevant, would be something like `std::fs::metadata(path)?.file_type().is_block_device()`. Let me try.

Reply:
Updated to use `std::fs:metadata` now.

Reply:
Nice :+1: